### ISSUE TITLE
cannot send specific text message

### ISSUE THREAD

**Issue by elandorr** [mattermost/mattermost-server#22479]
User cannot send a message, if it contains 'cookie banner'. The profanity filter is enabled, but set to replace filtered strings, so that's not it.

Logs are clean.

What the heck?

**Comment by amyblais** [comment]
Hi @elandorr,
What Mattermost server version are you on?
What OS and version are you on?

**Comment by elandorr** [comment]
Latest, running your Docker. 7.8.1 if I recall correctly. (just updated recently)

**Comment by amyblais** [comment]
@elandorr Do the users see an error message when they are not able to post the message?

**Comment by elandorr** [comment]
Nope, nothing. It swallows the message and that's it. Logs don't scream, either.

I tried a bunch of variations, 'cookie bannes' works, 'cookie banner' doesn't.

There are no invisible characters, I typed it myself.

What is it about that string?

**Comment by elandorr** [comment]
```
POST to posts
{
	"channel_id": "xxx",
	"create_at": 0,
	"file_ids": [],
	"message": "cookie banner",
	"metadata": {},
	"pending_post_id": "xxx:xxx",
	"props": {
		"disable_group_highlight": true
	},
	"reply_count": 0,
	"update_at": 1678488217896,
	"user_id": "xxx"
}

response
{
	"id": "xxx",
	"create_at": 1678488218051,
	"update_at": 1678488218051,
	"edit_at": 0,
	"delete_at": 0,
	"is_pinned": false,
	"user_id": "xxx",
	"channel_id": "xxx",
	"root_id": "",
	"original_id": "",
	"message": "cookie banner",
	"type": "",
	"props": {
		"disable_group_highlight": true
	},
	"hashtags": "",
	"pending_post_id": "xxx:xxx",
	"reply_count": 0,
	"last_reply_at": 0,
	"participants": null,
	"metadata": {}
}
```

Followed by two DELETE to drafts, both 'ok'.

**Comment by elandorr** [comment]
Turned out it was just that, sorry for bothering. I expected it couldn't have been some 'magic string' bug, didn't check all add-ons first time.

This gets rid of EU bs:
https://addons.mozilla.org/en-US/firefox/addon/i-dont-care-about-cookies/

But apparently this is not just a manually created set of filters, but also includes blanket searches such as 'cookie banner'. I did not expect that. It looks like it deletes the node, so nothing seems wrong, and it just disappears. Deleting a node like this seems silly, why would an actual banner have the string 'cookie banner' inside the element itself.
`<p>cookie banner</p>`

I won't bother reporting it to them, as this add-on has recently been taken over by a billion dollar corpo. If you use it, just disable it for your MM domain. Or better yet, uninstall it, as the corpos who bought it are well known for systematic spying and data abuse. They may start adding crap in an update without you knowing. I was personally just too annoyed by the EU bs to not use it. 

Alternatives I found:

- easy, activate the included 'EasyList Cookie' in uBo
- a decrappified fork of the other (didn't test) https://github.com/OhMyGuus/I-Still-Dont-Care-About-Cookies

Or this, which seems to be smarter about the actual dialogs, picking settings:

- https://github.com/cavi-au/Consent-O-Matic

That's probably smarter, although the EU bs is so idiotic, 99% of sites don't actually change behavior anyway (which is logical, as it's impractical). But this may choose the minimum in some cases. A bit rough with no export of the whitelist yet apparently, but seems neat.

It also explains how it does it, which is always nice.

Hope this helps someone running into the same issue.
